akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -78,8 +78,8 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
           case x: `Raw-Request-URI` ⇒ // we never render this header
             renderHeaders(tail, hostHeaderSeen, userAgentSeen, transferEncodingSeen)
 
-          case x: CustomHeader ⇒
-            if (!x.suppressRendering) render(x)
+          case x: CustomHeader if x.renderInRequests ⇒
```

Comment:
Thought about this move from inner if to guard for a while, but I agree. If we get `renderInRequests false` here then someone is doing it wrong and we don't need to optimise for that case.